### DISCUSSION TITLE
Merge eslint-plugin-rxjs into angular-eslint

### DISCUSSION THREAD

**Discussion by gultyayev** [angular-eslint/angular-eslint#2094] [New Rule Proposals]
It seems that the `eslint-plugin-rxjs` is no longer maintained. It would be nice if `angular-eslint` could merge these awesome rules to help developers writing better RxJS code.

**Comment by json-derulo** [comment]
For the rules from `eslint-plugin-rxjs-angular` it makes sense, but the `eslint-plugin-rxjs` is also used by non-Angular projects.

**Comment by biolauri** [comment]
Regarding `eslint-plugin-rxjs-angular`, this has already been asked at https://github.com/cartant/eslint-plugin-rxjs-angular/issues/23#issuecomment-2240799093. For `eslint-plugin-rxjs`, it's currently discussed at https://github.com/ReactiveX/rxjs/discussions/7492 to integrate it into rxjs itself.

IMHO, the split is the best thing to proceed:
* `eslint-plugin-rxjs` is best-fitting into `RxJS` itself as it has only general RxJS rules which is helpful for everyone using RxJS regardless their framework choices
* `eslint-plugin-rxjs-angular` is best-fitting here as it is specific to RxJS within Angular which is very common to do so

How shall we proceed? Convert this into an issue? Or do we need some feedback from one of the main contributors first?

**Comment by JasonWeinzierl** [comment]
I've started a fork https://github.com/JasonWeinzierl/eslint-plugin-rxjs-x that works with latest versions of ESLint and typescript-eslint.  I'm willing to maintain the fork as a permanent replacement for the original.

- But if the community decides to bring the rules into the RxJS monorepo, I'm happy to close the fork.  Ditto if the original author comes back/hands off the npm package/etc.
- I was planning on forking the rxjs-angular rules too, but incorporating into `angular-eslint` might be better.  ~No strong opinion on that.~ Actually yeah rolling it into `angular-eslint` would be nice because it could align with major versions of Angular, etc. and get better visibility to those rules instead of hiding in a niche separate package.  I personally had no clue the rxjs-angular rules existed until recently, despite using the rxjs rules for years.

**Reply by alfaproject** [reply]
@JasonWeinzierl do you still plan on creating `eslint-plugin-rxjs-angular-x`? It doesn't seem that the maintainers are interested in porting them here

**Reply by JasonWeinzierl** [reply]
I've published an initial fork of `eslint-plugin-rxjs-angular-x` on npm, from the repo here: https://github.com/JasonWeinzierl/eslint-plugin-rxjs-angular-x .  It (and the rxjs-x fork) is still prerelease as I review each rule and take the time to make any breaking changes before v1.0.0.  For example, those rules have no awareness of Angular's newish `takeUntilDestroyed` operator, so there's definitely room for improvement.

**Reply by JasonWeinzierl** [reply]
eslint-plugin-rxjs-x v1.0.0 is released with ESLint v10 support, 7 new rules, a strict config, and fixes for all outstanding bugs: https://github.com/JasonWeinzierl/eslint-plugin-rxjs-x/releases/tag/v1.0.0

The rxjs-angular-x fork is released as v1 too for ESLint v10 support.
